go-controller/pkg/ovn/egressip.go

```diff
@@ -2177,17 +2177,21 @@ func (e *egressIPController) addPodEgressIPAssignment(egressIPName string, statu
 			metrics.RecordEgressIPAssign(duration)
 		}()
 	}
-	if err = e.deleteExternalGWPodSNAT(pod, podIPs, status); err != nil {
+	ops, err := e.deleteExternalGWPodSNATOps(nil, pod, podIPs, status)
+	if err != nil {
 		return err
 	}
-	if err = e.createReroutePolicy(podIPs, status, egressIPName); err != nil {
-		return fmt.Errorf("unable to create logical router policy, err: %v", err)
-	}
-	var ops []ovsdb.Operation
-	ops, err = createNATRuleOps(e.nbClient, podIPs, status, egressIPName)
+
+	ops, err = createNATRuleOps(e.nbClient, ops, podIPs, status, egressIPName)
 	if err != nil {
 		return fmt.Errorf("unable to create NAT rule for status: %v, err: %v", status, err)
 	}
+
+	ops, err = e.createReroutePolicyOps(ops, podIPs, status, egressIPName)
+	if err != nil {
+		return fmt.Errorf("unable to create logical router policy, err: %v", err)
+	}
+
```

Comment:
self-note: 1) we are deleting SNAT towards nodeIP for the pod 2) we are adding SNAT towards egressIP of the pod 3) we are adding rereoute policy for the pod

nit: would be nice to outline in the commit message (and update PR description so that anyone who doesn't have context can understand what ops you are talking about) the fact that these are the 3 ops we are combining together and that adding SNAT will be done before adding reroute to avoid that small air gap. - this will set stage for your other PR as well